projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLUtils.java

```diff
@@ -48,4 +51,31 @@ public static void validatePageInput(int pageSize, int pageNumber) {
       }
    }
 
+    /**
+     * This method counts Failed and Finished statuses for
+     * all data job executions for a given data job and
+     * returns an immutable pair of fail and success counts.
+     *
+     * @param dataJobName            the data job name.
+     * @param jobExecutionRepository the repository to query.
+     * @return ImmutablePair<Integer, Integer>, the left element is the Fail count, right Finished.
+     */
+    public static ImmutablePair<Integer, Integer> countFailedAndFinishedExecutions(String dataJobName,
```

Comment:
Why not list of job names? We want to execute one query that fetches the data for all data jobs rather than one query per data job.